### PR TITLE
Update regex for `looselyParseAmount` for 5-9 decimal places

### PR DESCRIPTION
**Pull request by sreetamdas**
Update the regex to parse 5-9 decimal places, instead of just 5.

---

While trying to import transactions using a CSV file (that I download from my stock broker) where the amounts contain 6 decimal places, I encountered a bug where the amount was incorrectly parsed: `15.930000` as `15930000` 😅 

This is the code block that parses amounts when importing transactions ([link](https://github.com/actualbudget/actual/blob/b803731bc52dbf3628b2129c7881cebabd24488e/packages/loot-core/src/shared/util.ts#L404-L414)):

```ts
function extractNumbers(v: string): string {
  // decimal is removed, 15.930000 ➡️ 15930000
  return v.replace(/[^0-9-]/g, "");
}

// look for a decimal marker, then look for either 5 or 1-2 decimal places.
// This avoids matching against 3 places which may not actually be decimal
const m = amount.match(/[.,]([^.,]{5}|[^.,]{1,2})$/);
if (!m || m.index === undefined || m.index === 0) {
  return safeNumber(parseFloat(extractNumbers(amount)));
}
```

### CODE REVIEW

**Review by youngcw — APPROVED** [review]
Thanks.  Ill wait until after the merge freeze to merge this in.

File: packages/loot-core/src/shared/util.test.ts
Comment (psybers) [inline packages/loot-core/src/shared/util.test.ts:10]: There is a comment a few lines above this that should be updated for the new parsing.

**Review by psybers — CHANGES_REQUESTED** [review]

**Review by psybers — APPROVED** [review]

**Comment by youngcw** [comment]
Can you look into why VRT is failing?  Double check that you are fully up to date with master, but it could still be an external problem

**Review by youngcw — APPROVED** [review]

**Pull request merged**
